libdrizzle/pack.cc

```diff
@@ -481,3 +481,24 @@ static drizzle_return_t _pack_scramble_hash(drizzle_st *con,
 
   return DRIZZLE_RETURN_OK;
 }
+
+void drizzle_unpack_error(drizzle_st *con)
+{
+  assert(con->buffer_ptr[0] == 255);
```

Comment:
I'd say make it `drizzle_check_unpack_error` and return `false` if it's `!= 255`.
This way you loose the assert, and you can use it in `if`. Also less usage of the magic `255`.